File: Testing/Code for Testing/Early Basic Syntax Testing/fixed_enum/test_fixed_enum.js
const Priority = Object.freeze({
    low: 1, medium: 2, high: 3,
    _names: {[1]: "low", [2]: "medium", [3]: "high"},
    _values: new Set([1, 2, 3]),
    name(val) { return this._names[val] || "unknown"; },
    valid(val) { return this._values.has(val); }
});

function main() {
    let p = Priority.high;
    console.log(p);
    let q = Priority.low;
    console.log(q);
    if ((p > q)) {
        console.log("high > low");
    }
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/fixed_enum/test_fixed_enum.py
from dataclasses import dataclass, field

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

from enum import IntEnum

class Priority(IntEnum):
    low = 1
    medium = 2
    high = 3


def main():
    p = Priority.high
    print(p)
    q = Priority.low
    print(q)
    if (p > q):
        print('high > low')
    return 0


if __name__ == "__main__":
    main()
